### PR TITLE
Implement static compression and encryption pipeline

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**
This builds up on the static sizer and timeout PR: https://github.com/GeyserMC/MCProtocolLib/pull/833
The reason compression and encryption should be static is to avoid confusion about the pipeline dynamically changing. This also allows more customization by developers who want to use alternative compression algorithms like igzip.

### CODE REVIEW

File: protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java
Comment (onebeastchris) [inline protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java]: ?

File: protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java
Comment (onebeastchris) [inline protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java]:
```suggestion
     * Sets the compression config for this session.
```

File: protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java
Comment (onebeastchris) [inline protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java]:
```suggestion
     * @param compressionConfig the compression to compress with,
```

File: protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java
Comment (onebeastchris) [inline protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java]:
```suggestion
     * @param encryptionConfig the encryption to encrypt with,
```

**Review by onebeastchris — COMMENTED** [review]
minor notes; looks okay to me otherwise

File: protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java
Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java]: no idea how that happened

**Review by onebeastchris — APPROVED** [review]
LGTM, but I'll wait on others to review too before merging

File: protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java
Comment (Konicai) [inline protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java]: nullable annotation, same w encryption setter, and apply to TcpSession as well

File: protocol/src/main/java/org/geysermc/mcprotocollib/network/tcp/TcpSession.java
Comment (Konicai) [inline protocol/src/main/java/org/geysermc/mcprotocollib/network/tcp/TcpSession.java]:
```suggestion
            throw new IllegalStateException("You need to connect to set the compression!");
```
consistency with setEncryption?

File: protocol/src/main/java/org/geysermc/mcprotocollib/protocol/ClientListener.java
Comment (Konicai) [inline protocol/src/main/java/org/geysermc/mcprotocollib/protocol/ClientListener.java]: save threshold to a variable just to make it more readable

**Comment by Konicai** [comment]
nice btw :)

File: protocol/src/main/java/org/geysermc/mcprotocollib/protocol/ClientListener.java
Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/protocol/ClientListener.java]: done

File: protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java
Comment (AlexProgrammerDE) [reply protocol/src/main/java/org/geysermc/mcprotocollib/network/Session.java]: done

File: protocol/src/main/java/org/geysermc/mcprotocollib/protocol/MinecraftProtocol.java
Comment (Konicai) [inline protocol/src/main/java/org/geysermc/mcprotocollib/protocol/MinecraftProtocol.java]: would be nice to take this opportunity to rename the method and exception message from `enable` -> `create`

**Review by Konicai — APPROVED** [review]

**Pull request merged**
